sympy/printing/latex.py

```diff
@@ -1721,6 +1720,19 @@ def _print_MatMul(self, expr):
         else:
             return ' '.join(map(parens, args))
 
+    def _print_Determinant(self, expr, exp=None):
+        if expr.args[0].is_MatrixExpr:
+            tex = r"\left|{%s}\right|" % self._print(expr.args[0])
+        else:
+            mat_delim_backup = self._settings['mat_delim']
+            self._settings['mat_delim'] = ''
+            tex = r"\left|{%s}\right|" % self._print(expr.args[0])
+            self._settings['mat_delim'] = mat_delim_backup
+        if exp is not None:
+            return r"%s^{%s}" % (tex, exp)
```

Comment:
The Pow printer only uses the exp argument if the argument is a Function. Maybe this could be generalized to allow other classes, although in this case, do we really want powers of determinants to print like `|A|^2`? A test you added below seems to indicate not. Personally I'd say we should not print determinant powers like that because it makes it unclear if an expression is actually `Determinant(MatPow)` or `Pow(Determinant)`.

Reply:
The plan was that it should be printed as `|A|^2`, yes. And the other way around as `|A^2|` (or `|(A)^2|` if A is a `Matrix` rather than a `MatrixSymbol`).

Probably that was a bit too much copy and paste...